Fix EagerLoadPolyAssocsTest setup

alimi opened the pull request:
### Summary

`EagerLoadPolyAssocsTest` includes a `Remembered` module in multiple test ActiveRecord classes. The module is supposed to keep track of records created for each of the included classes individually, but it collects all records for every class. This happens because `@@remembered` is defined on the `Remembered` module and shared between the ActiveRecord classes.

We can see this with a test like

```ruby
assert_equal NUM_SIMPLE_OBJS, Circle.remembered.length
```

`Circle.remembered.length` should only be 50 (`NUM_SIMPLE_OBJS`), but it's 350 because there are `NUM_SIMPLE_OBJS` being created for 7 AR classes.

This only becomes an issue for databases (like CockroachDB) that use [random primary keys instead of sequential ones by default](https://www.cockroachlabs.com/docs/v19.2/serial.html#modes-of-operation).

To fix the bug, we can make the remembered collection name unique per ActiveRecord class.

### Other Information

I'm not sure this is the best way to fix the test, but it was the path of least resistance. I'm open to suggestions! 😄

alimi committed:
Fix EagerLoadPolyAssocsTest setup

  * EagerLoadPolyAssocsTest includes a Remembered module in multiple test
    ActiveRecord classes. The module is supposed to keep track of records
    created for each of the included classes individually, but it collects all
    records for every class. This happens because @@remembered is defined on the
    Remembered module and shared between the ActiveRecord classes.  This only
    becomes an issue for databases (like CockroachDB) that use random primary
    keys instead of sequential ones by default.
  * To fix the bug, we can make the remembered collection name unique per
    ActiveRecord class.

jonathanhefner commented on activerecord/test/cases/associations/eager_load_nested_include_test.rb:
```diff
@@ -20,8 +20,17 @@ def remember; self.class.remembered << self; end
   module ClassMethods
-    def remembered; @@remembered ||= []; end
-    def sample; @@remembered.sample; end
+    def remembered
```
This should also work:
```diff
-    def remembered; @@remembered ||= []; end
+    def remembered; @remembered ||= []; end
```

jonathanhefner commented:
It took me a little while to understand why this test is broken only for non-sequential IDs.  I finally realized: it's always been broken!  It's just that sequential IDs happen to prevent the test from failing, because an ID of "1" is an ID of "1" regardless of model class.

I left a comment on how to simplify the solution a bit.  Thank you for discovering this!

alimi committed:
Update EagerLoadPolyAssocsTest test setup

  * Instead of defining remembered as a class variable, we can define it as an
    instance variable that will be unique to every class that includes the
    Remembered module.
